### ISSUE TITLE
`normalize_transformation` divides by `M[2,2] + eps`: the last entry becomes 0.990 for 1e-6 and 101 for -1.01e-8

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4874]
`normalize_transformation(M)` is documented to scale `M` "so that the value in the last row and column is one", and `find_fundamental` returns its result for both methods (`fundamental.py:253`, `:326`). It computes `torch.where(|M[2,2]| > eps, M / (M[2,2] + eps), M)` (`fundamental.py:102`, `eps = 1e-8`), so the last entry comes out as `M[2,2] / (M[2,2] + eps)`: 0.990 for `M[2,2] = 1e-6`, 1.11 for `-1e-7`, 0.502 just above the threshold, and 101 for `-1.01e-8`, where `eps` moves a negative divisor towards zero. Just below the threshold the matrix is returned unscaled, so the output jumps from 0.5 to 1e-8 across it. The `torch.where` guard already excludes the singular case, so the `+ eps` changes the value and guards nothing; the expected result is `M / M[2,2]` wherever the guard admits the division.

## Repro

```python
import torch
from kornia.geometry.epipolar import find_fundamental, normalize_transformation

# normalize_transformation promises "the value in the last row and column is one"
for v in [1.0, 1e-3, 1e-6, -1e-7, 1.01e-8, 0.99e-8, -1.01e-8]:
    M = torch.tensor([[2.0, 0.5, 1.0], [0.3, 1.5, -2.0], [0.1, 0.2, v]])
    print(f"M[2,2] = {v:<9g} -> out[2,2] = {normalize_transformation(M)[2, 2].item():.6g}")

# find_fundamental ends with normalize_transformation. Rectified stereo (same K, pure x baseline) has a
# true F[2,2] of 0, so in float32 the raw entry before the division is rounding noise of order 1e-6.
K = torch.tensor([[800.0, 0, 320], [0, 800, 240], [0, 0, 1]], dtype=torch.float64)
torch.manual_seed(0)
X = torch.rand(20, 3, dtype=torch.float64) * 2 - 1
X[:, 2] += 5
def proj(t):
    x = (K @ (X.T + t)).T
    return (x[:, :2] / x[:, 2:])[None].float()
F = find_fundamental(proj(torch.zeros(3, 1, dtype=torch.float64)), proj(torch.tensor([[-1.0], [1e-6], [0.0]], dtype=torch.float64)))
print("find_fundamental F[2,2] =", F[0, 2, 2].item())
```

Output:

```text
M[2,2] = 1         -> out[2,2] = 1
M[2,2] = 0.001     -> out[2,2] = 0.99999
M[2,2] = 1e-06     -> out[2,2] = 0.990099
M[2,2] = -1e-07    -> out[2,2] = 1.11111
M[2,2] = 1.01e-08  -> out[2,2] = 0.502488
M[2,2] = 9.9e-09   -> out[2,2] = 9.9e-09
M[2,2] = -1.01e-08 -> out[2,2] = 101
find_fundamental F[2,2] = 0.9916811585426331
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same, except the last line: `find_fundamental F[2,2] = 0.9995392560958862`), CPU float32, macOS arm64. In the last line the true `F[2,2]` of a rectified pair is 0, so the entry before the division is float32 rounding noise, and `0.99168 = r / (r + 1e-8)` for a raw entry `r` of about 1.2e-6. That `F[2,2] = 1` is itself an ill-posed scale for such a pair is a separate question; this issue is the `+ eps` only.

`find_homography_dlt` and `find_homography_lines_dlt` end with the same expression, `H / (H[..., -1:, -1:] + eps)` (`homography.py:291`, `:428`), without the `torch.where` guard.

## Classification

Silent wrong result. Idiom 1 (`x + eps` perturbing a result instead of guarding a singularity).

## Disposition

Focused fix, welcome as a PR: divide by the entry itself behind the existing guard, with a safe divisor so the branch `torch.where` discards stays finite for autograd, e.g. `d = torch.where(norm_val.abs() > eps, norm_val, torch.ones_like(norm_val))` and `torch.where(norm_val.abs() > eps, M / d, M)`; apply the same to `homography.py:291` and `:428`. At the usual scale (`|M[2,2]|` of order 1) the output changes by about `eps`, relative.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #4881 (same idiom in the epipolar and homography metrics).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
